pool.go

```diff
@@ -47,6 +52,16 @@ type ClientConn struct {
 // clients could not be created
 func New(factory Factory, init, capacity int, idleTimeout time.Duration,
 	maxLifeDuration ...time.Duration) (*Pool, error) {
+	return NewWithContext(context.TODO(), func(ctx context.Context) (*grpc.ClientConn, error) { return factory() },
```

Comment:
TODO is meant to be used in parts of codes that should be updated with appropriate context in the future. In this case we won't have any way of retrieving the context since this was out of the API, so let's just leave it as Background for now.

```suggestion
	return NewWithContext(context.Background(), func(ctx context.Context) (*grpc.ClientConn, error) { return factory() },
```

Reply:
agree, changed.